### PR TITLE
[RAISE-BP] Check -triton-raise-block-pointer output leads to 2D block ops

### PR DESCRIPTION
**Pull request by mfrancepillois**
Add lit test checking that the raised code is lowered using 2d block load operations.

Closes Issue: #1431

### CODE REVIEW

**Review by victor-eds — APPROVED** [review]
This is great! I assume this optimization does not kick in without the pass, right?

File: test/TritonIntelGPU/raise-block-pointer-gen-code.mlir [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// RUN: triton-opt %s -triton-raise-block-pointer -canonicalize  -convert-triton-to-tritongpu=target="xpu:DEVICE_ARCH.PVC threads-per-warp=16" -tritongpu-remove-layout-conversions  --tritonintelgpu-accelerate-matmul --tritonintelgpu-remove-layout-conversions --tritonintelgpu-rewrite-tensor-pointer --intel-decompose-unsupported-conversions --convert-scf-to-cf --convert-index-to-llvm --convert-arith-to-llvm --canonicalize --cse --symbol-dce --intel-allocate-shared-memory --convert-triton-intel-gpu-to-llvm | FileCheck %s
```
Comment (Dewei-Wang-sh) [inline test/TritonIntelGPU/raise-block-pointer-gen-code.mlir:1]: is this appropriate for a unit test?

Comment (mfrancepillois) [reply test/TritonIntelGPU/raise-block-pointer-gen-code.mlir:1]: It's true that this test need several passes to check the generated code as the raising pass is one of the first in the pipeline.
However, other lit tests, such as https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/test/TritonIntelGPU/load-to-llvm-2dload.mlir, invoke two passes.

Comment (etiotto) [reply test/TritonIntelGPU/raise-block-pointer-gen-code.mlir:1]: I think lit test should test individual transformations. So, for the most part, they should limit themselves to run an individual pass. 

For integration tests we have the python tests.

Comment (whitneywhtsang) [reply test/TritonIntelGPU/raise-block-pointer-gen-code.mlir:1]: Some e2e tests, e.g., test_core.py, check for generated ptx, can we do something similar to test that 2d block ops are generated on xpu for the gemm tutorials?

File: third_party/intel/lib/TritonRaiseBlockPointer/TritonRaiseBlockPointer.cpp [hunk 2]
```diff
@@ -50,6 +50,30 @@ struct PtrState {
     return offsets.size();
   }
 
+  bool isBlockPtr() const { return !order.empty(); }
```
[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonRaiseBlockPointer/TritonRaiseBlockPointer.cpp]: done

**Comment by etiotto** [comment]
@mfrancepillois can we close this one ?

**Comment by sommerlukas** [comment]
> @mfrancepillois can we close this one ?

@mfrancepillois is off, but our plan was to close this PR once we have implemented the alternative that we discussed offline.

**Pull request closed without merging**